### PR TITLE
Melodic devel a1 adaptation

### PR DESCRIPTION
**Pull request by jrenaf**
This PR contains: 
1.  Add a1 description including xacro, urdf and sdf
2. Put robot type as an argument in roslaunch
3. a1 is able to run on rough terrain

### CODE REVIEW

File: local_planner/include/local_planner/local_planner.h [hunk 1]
```diff
@@ -99,6 +99,9 @@ class LocalPlanner {
    */
   void publishLocalPlan();
 
+  /// Robot type: A1 or Spirit
+  std::string robot_;
```
Comment (ardalantj) [inline local_planner/include/local_planner/local_planner.h:103]: robot_name_?

File: local_planner/src/local_planner.cpp [hunk 2]
```diff
@@ -116,7 +119,17 @@ LocalPlanner::LocalPlanner(ros::NodeHandle nh)
 
 void LocalPlanner::initLocalBodyPlanner() {
   // Create nmpc wrapper class
-  local_body_planner_nonlinear_ = std::make_shared<NMPCController>(0);
+  int type;
```
Comment (ardalantj) [inline local_planner/src/local_planner.cpp:122]: This can be an enum

File: nmpc_controller/include/nmpc_controller/nmpc_controller.h [hunk 3]
```diff
@@ -86,6 +86,9 @@ class NMPCController {
  private:
   ros::NodeHandle nh_;
 
+  /// Robot type: A1 or Spirit
+  std::string robot_;
```
Comment (ardalantj) [inline nmpc_controller/include/nmpc_controller/nmpc_controller.h:90]: robot_name_?

File: quad_simulator/a1_description/xacro/gazebo.xacro [hunk 4]
```diff
@@ -0,0 +1,324 @@
...
+        <plugin name="gazebo_ros_control" filename="libgazebo_ros_control.so">
+            <robotNamespace>/a1</robotNamespace>
+            <robotSimType>gazebo_ros_control/DefaultRobotHWSim</robotSimType>
+        </plugin>
+        <plugin filename="libground_truth_estimator.so" name="ground_truth_estimator">
+            <updateRateHZ>500.0</updateRateHZ>
+        </plugin>
+    </gazebo>
+    <!-- Show the trajectory of trunk center. -->
+    <!-- <gazebo>
```
Comment (ardalantj) [inline quad_simulator/a1_description/xacro/gazebo.xacro:14]: Remove the commented section?

File: quad_simulator/a1_description/xacro/gazebo.xacro [hunk 5]
```diff
@@ -0,0 +1,324 @@
...
+                <gaussianNoise>0.0</gaussianNoise>
+                <xyzOffset>0 0 0</xyzOffset>
+                <rpyOffset>0 0 0</rpyOffset>
+                <frameName>imu_link</frameName>
+            </plugin>
+            <pose>0 0 0 0 0 0</pose>
+        </sensor>
+    </gazebo>
+
+    <!-- Foot contacts. -->
```
Comment (ardalantj) [inline quad_simulator/a1_description/xacro/gazebo.xacro:64]: Remove?

File: robot_driver/src/robot_driver.cpp [hunk 6]
```diff
@@ -129,7 +134,7 @@ RobotDriver::RobotDriver(ros::NodeHandle nh, int argc, char** argv) {
   last_state_time_ = std::numeric_limits<double>::max();
 
   // Set joint torque limits
-  torque_limits_ << 21, 21, 32;
+  // torque_limits_ << 21, 21, 32;
```
Comment (ardalantj) [inline robot_driver/src/robot_driver.cpp:137]: remove?

File: quad_simulator/a1_description/xacro/gazebo.xacro (hunk 4 above)
Comment (jrenaf) [reply quad_simulator/a1_description/xacro/gazebo.xacro:14]: We might want to leave it there first, that's used by the original a1 description from unitree

Comment (jrenaf) [reply quad_simulator/a1_description/xacro/gazebo.xacro:14]: We can make decision to leave it or not during refactoring.

File: local_planner/include/local_planner/local_planner.h (hunk 1 above)
Comment (jrenaf) [reply local_planner/include/local_planner/local_planner.h]: solved

File: local_planner/src/local_planner.cpp (hunk 2 above)
Comment (jrenaf) [reply local_planner/src/local_planner.cpp]: solved

File: nmpc_controller/include/nmpc_controller/nmpc_controller.h (hunk 3 above)
Comment (jrenaf) [reply nmpc_controller/include/nmpc_controller/nmpc_controller.h:90]: solved

File: quad_simulator/a1_description/xacro/gazebo.xacro (hunk 5 above)
Comment (jrenaf) [reply quad_simulator/a1_description/xacro/gazebo.xacro:64]: same as above

**Comment by jcnorby** [comment]
These changes will be included in the multi-robot PR.

**Pull request closed without merging**